test/jdk/sun/security/ssl/SSLEngineImpl/TestBadDNForPeerCA.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8294985
+ * @library /test/lib
+ * @summary SSLEngine throws IAE during parsing of X500Principal
+ * @run main/othervm TestBadDNForPeerCA
```

Comment:
```suggestion
 * @run main/othervm TestBadDNForPeerCA
 * @run main/othervm -Djavax.net.debug=all TestBadDNForPeerCA
```
and then remove the `debug` field.

Reply:
When we put in the debug field into the template, it was to allow folks to quickly add debugging output to their test, but in general it won't be turned on during regular test runs.

Reply:
I'd like to enable this in the test runs as well; otherwise the changes to `toString` method will not be tested.

Reply:
Ah!  Thanks for clarifying.